apiserver/src/main/java/org/dependencytrack/resources/v1/UserResource.java

```diff
@@ -775,4 +784,64 @@ private UserSubject buildUserSubject(final String username, final String email)
+    @Produces(MediaType.APPLICATION_JSON)
+    @Operation(summary = "Adds the username to the specified team.", description = "<p>Requires permission <strong>ACCESS_MANAGEMENT</strong> or <strong>ACCESS_MANAGEMENT_UPDATE</strong></p>")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "The updated user", content = @Content(schema = @Schema(implementation = UserPrincipal.class))),
+            @ApiResponse(responseCode = "304", description = "The user is already a member of the specified team(s)"),
+            @ApiResponse(responseCode = "401", description = "Unauthorized"),
+            @ApiResponse(responseCode = "404", description = "The user or team(s) could not be found")
+    })
+    @PermissionRequired({ Permissions.Constants.ACCESS_MANAGEMENT, Permissions.Constants.ACCESS_MANAGEMENT_UPDATE })
+    public Response setUserTeams(
+            @Parameter(description = "A valid username", required = true) @PathParam("username") String username,
+            @Parameter(description = "The UUID(s) of the team(s) to associate username with", required = true) @Valid TeamsSetRequest request) {
+        try (QueryManager qm = new QueryManager()) {
+            UserPrincipal principal = qm.getUserPrincipal(username);
+            if (principal == null) {
+                return Response.status(Response.Status.NOT_FOUND).entity("The user could not be found.").build();
+            }
+
+            // Compare given team uuids against current user teams
+            final Set<String> currentUserTeams = principal.getTeams() == null ? Collections.emptySet()
+                    : principal.getTeams().stream()
+                            .map(Team::getUuid)
+                            .map(UUID::toString)
+                            .collect(Collectors.toSet());
+
+            if (currentUserTeams.equals(request.getTeams())) {
+                return Response.notModified().entity("The user is already a member of the selected team(s)").build();
+            }
+
+            List<Team> requestedTeams = request.getTeams()
+                    .stream()
+                    .map(uuid -> qm.getObjectByUuid(Team.class, uuid))
+                    .toList();
+
+            // check that all requested teams exist
+            List<String> notFound = new ArrayList<>();
+            for (int i = 0; i < requestedTeams.size(); i++) {
+                if (requestedTeams.get(i) == null)
+                    notFound.add(request.getTeams().stream().toList().get(i));
+            }
+
+            if (notFound.size() > 0) {
+                // String msg = String.format("One or more teams could not be found:\n%s", );
+                Map<String, Object> response = new HashMap<>();
+                response.put("error", "One or more teams could not be found");
+                response.put("teams", notFound);
+                return Response.status(Response.Status.BAD_REQUEST).entity(response).build();
+            }
```

Comment:
Same as the other endpoint, have a look at [`ProblemDetails`](https://github.com/DependencyTrack/hyades-apiserver/blob/0b800bdb51fadb28e0c9929357791268615f8a23/apiserver/src/main/java/org/dependencytrack/resources/v1/problems/ProblemDetails.java).